driver/bpf/fillers.h

```diff
@@ -4845,6 +4845,166 @@ FILLER(sched_drop, false)
 	return bpf_push_u32_to_ring(data, data->settings->sampling_ratio);
 }
 
+/* In this kernel version the instruction limit was bumped to 1000000 */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 2, 0))
+#define MAX_THREADS_GROUPS 30
+#define MAX_HIERARCHY_TRAVERSE 60
+#else
+/* We need to find the right calibration here. On kernel 4.14 the limit
+ * seems to be MAX_THREADS_GROUPS*MAX_HIERARCHY_TRAVERSE <= 100
+ */
+#define MAX_THREADS_GROUPS 10
+#define MAX_HIERARCHY_TRAVERSE 10
+#endif
+
+/* 3 possible cases:
+ * - Looping between all threads of the current thread group we don't find a valid reaper. -> return 0
+ * - We cannot loop over all threads of the group due to BPF verifier limits (MAX_THREADS_GROUPS) -> return -1
+ * - We find a reaper -> return its `pid`
+ */
+static __always_inline pid_t find_alive_thread(struct task_struct *father)
+{
+	struct signal_struct *signal = (struct signal_struct *)_READ(father->signal);
+	struct list_head *head = &(signal->thread_head);
+	struct list_head *next_thread = (struct list_head *)_READ(head->next);
+
+	u8 cnt = 0;
+	u32 flags = 0;
+
+#pragma unroll MAX_THREADS_GROUPS
+	for(struct task_struct *t = container_of(next_thread, typeof(struct task_struct), thread_node);
+	    next_thread != (head) && cnt < MAX_THREADS_GROUPS;
+	    t = container_of(next_thread, typeof(struct task_struct), thread_node))
+	{
+		cnt++;
+		bpf_probe_read_kernel(&flags, sizeof(flags), &t->flags);
+		if(!(flags & PF_EXITING))
+		{
+			/* Found it */
+			return _READ(t->pid);
+		}
+		next_thread = (struct list_head *)_READ(t->thread_node.next);
+	}
+
+	/* If we cannot loop over all threads, we cannot know the right reaper */
+	if(cnt == MAX_THREADS_GROUPS)
+	{
+		return -1;
+	}
+
+	/* We didn't find it */
+	return 0;
+}
+
+/* When we die, we re-parent all our children, and try to:
+ * 1. give them to another thread in our thread group, if such a member exists
+ * 2. give it to the first ancestor process which prctl'd itself as a
+ *    child_subreaper for its children (like a service manager)
+ * 3. give it to the init process (PID 1) in our pid namespace
+ */
+static __always_inline pid_t find_new_reaper_pid(struct filler_data *data, struct task_struct *father)
+{
+	pid_t reaper_pid = find_alive_thread(father);
+	
+	/* - If we are not able to find the reaper due to BPF
```

Comment:
as I wrote in the PR description if we are not able to find a valid reaper due to the verifier limit we will tell the userspace to use its custom logic sending `-1`